spec/builtin-trap.test.sh

```diff
@@ -288,3 +288,14 @@ on exit
 status=0
 ## END
 
+#### Remove trap with an unsigned integer
+
+trap 'echo noprint' EXIT
+trap 1 EXIT
+echo printed
```

Comment:
One thing I realized is that we're not respecting multiple traps - `trap 'echo hi' INT EXIT`

But that's orthogonal to this change, so we can leave it out for now



```
$ bash -c 'trap "echo hi" EXIT INT; trap -p; echo ; trap "echo z" INT; trap -p; trap 1 EXIT INT; trap -p'
trap -- 'echo hi' EXIT
trap -- 'echo hi' SIGINT

trap -- 'echo hi' EXIT
trap -- 'echo z' SIGINT
```

Reply:
Yeah, I noticed that too when linking to the spec; should I open an issue (and/or make a failing spec test) or are you going to?

Reply:
A failing spec test would be appreciated!

and then update `oils_failures_allowed:` at the top of the file